runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -169,10 +178,27 @@ else if (client.getGameState() != GameState.LOGGED_IN)
 		graphics.fill(new Rectangle(client.getCanvas().getSize()));
 		graphics.setColor(color);
 
-		if (Instant.now().minusMillis(FLASH_DURATION).isAfter(flashStart))
+		// Enforce a minimum flash duration of 2 seconds so as not to cancel it too quickly
+		if (!Instant.now().minusMillis(MINIMUM_FLASH_DURATION_MILLIS).isAfter(flashStart))
```

Comment:
This whole block is pretty confusing and not very straightforward. I would recommend rewriting it to be more like the following so it is easier to differentiate between the guard clause and the code which is actually updating the flash timer.

```java
if (!Instant.now().minusMillis(MINIMUM_FLASH_DURATION_MILLIS).isAfter(flashStart))
{
    return;
}

switch (notificationFlashType)
{
	case FLASH_TWO_SECONDS:
	case SOLID_TWO_SECONDS:
		flashStart = null;
		break;
	case SOLID_UNTIL_CANCELLED:
	case FLASH_UNTIL_CANCELLED:
	{
		if (client.getMouseIdleTicks() < MINIMUM_FLASH_DURATION_TICKS
			|| client.getKeyboardIdleTicks() < MINIMUM_FLASH_DURATION_TICKS
			|| client.getMouseLastPressedMillis() > mouseLastPressedMillis)
		{
            flashStart = null;
		}
		break;
	}
}
```

Reply:
Yea this is much a clearer way of doing it, thanks.